acceptance/tests/parser_functions/hiera_in_templates.rb

```diff
@@ -0,0 +1,403 @@
+test_name "Calling Hiera function from inside templates"
+
+# Global variables
+
+$module_name = "hieratest"
+$coderoot = master.tmpdir("#{$module_name}")
+$resultdir = "#{$coderoot}/results"
+
+$msg_default = 'message from default.yaml'
+$msg_production = 'message from production.yaml'
+$msg1os = 'message1 from {osfamily}.yaml'
+$msg2os = 'message2 from {osfamily}.yaml'
+$msg_fqdn = 'messsage from {fqdn}.yaml'
+
+$k1 = 'key1'
+$k2 = 'key2'
+$k3 = 'key3'
+
+$val1p = 'value1 from production.yaml'
+$val2p = 'value2 from production.yaml'
+$val2os = 'value2 from {osfamily}.yaml'
+$val3os = 'value3 from {osfamily}.yaml'
+
+$hval3os = 'hash_value3 from production.yaml'
+$hval2os = 'hash_value2 from production.yaml'
+$hval1p = 'hash_value1 from {osfamily}.yaml'
+$hval2p = 'hash_value2 from {osfamily}.yaml'
+
+$h_m_call = "hiera\\('message'\\)"
+$h_h_call = "hiera\\('hash_value'\\)"
+$h_i_call = "hiera\\('includes'\\)"
+$ha_m_call = "hiera_array\\('message'\\)"
+$ha_i_call = "hiera_array\\('includes'\\)"
+$hh_h_call = "hiera_hash\\('hash_value'\\)"
+
+$mod_default_msg = 'This file created by mod_default.'
+$mod_osfamily_msg = 'This file created by mod_osfamily.'
+$mod_production_msg = 'This file created by mod_production.'
+$mod_fqdn_msg = 'This file created by mod_fqdn.'
+
+
+def create_environment
+
+  #    Creates a directory tree like this on the master, under /tmp/hieratest/
```

Comment:
we should probably remove this comment.  most people around here assume a tree like this and the rest can be inferred from the puppet code "as documentation".